src/client/components/GoogleLoginButton/GoogleLoginButton.component.js

```diff
@@ -1,14 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './GoogleLoginButton.styles.css';
 import GoogleLogo from '../../assets/images/Google_logo.png';
+import { useFirebase } from '../../firebase/FirebaseContext';
+import Loader from '../Loader/index';
+import { useHistory } from 'react-router-dom';
+
+// import { auth } from 'firebase';
 
 const GoogleLoginButton = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [userName, setUserName] = useState('');
+  const history = useHistory();
+
+  const { auth, signInGoogle, googleSignOut } = useFirebase();
+
+  const onClick = async () => {
+    setIsLoading(true);
+    const name = await signInGoogle();
```

Comment:
So sign in returns the name?

Reply:
Yes, In signIn function , i returned user.displayname, and storing in the name